viewer/packages/cad-model/src/batching/MultiBufferBatchingManager.ts

```diff
@@ -0,0 +1,313 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { Materials } from '@reveal/rendering';
+import { ParsedGeometry, RevealGeometryCollectionType } from '@reveal/sector-parser';
+import {
+  decrementOrDeleteIndex,
+  DynamicDefragmentedBuffer,
+  incrementOrInsertIndex,
+  TypedArray
+} from '@reveal/utilities';
+import { BufferGeometry, Group, InstancedMesh, InterleavedBufferAttribute, RawShaderMaterial } from 'three';
+import { GeometryBufferUtils } from '../utilities/GeometryBufferUtils';
+import { getShaderMaterial } from '../utilities/getShaderMaterial';
+import { DrawCallBatchingManager } from './DrawCallBatchingManager';
+
+type SectorBatch = {
+  instanceBatches: Map<string, SectorInstanceData>;
+};
+
+type SectorInstanceData = {
+  batchBuffer: BatchBuffer;
+  batchId: number;
+  instanceCount: number;
+};
+
+type InstanceBatch = {
+  buffers: BatchBuffer[];
+};
+
+type BatchBuffer = {
+  mesh: InstancedMesh;
+  buffer: DynamicDefragmentedBuffer<Uint8Array>;
+};
```

Comment:
Also, I believe it would be helpful to document what these are. I'm thinking just comments explaining what each type is used for. Also - took me a while to understand that "instance" means "primitive type" in most cases, which could also be helpful to document